src/biotite/structure/transform.py

```diff
@@ -234,6 +234,105 @@ def rotate_about_axis(atoms, axis, angle, support=None):
     return _put_back(atoms, positions)
 
 
+def orient_principal_components(atoms, order=None):
+    """
+    Translate and rotate the atoms to be centered at the origin with
+    the principal axes aligned to the Cartesian axes, as specified by
+    the order keyword. By default, x, y, z.
+
+    By default, the resulting coordinates have the highest variance in
+    the x-axis and the lowest variance on the z-axis. Setting the order
+    keyword argument will change the direction of the highest variance.
+    For example, order=(2, 1, 0) results in highest valianxe along the
+    z-axis and lowest along the x-axis.
+
+    Parameters
+    ----------
+    atoms : AtomArray or ndarray, shape=(n,3)
+        The atoms of which the coordinates are transformed.
+        The coordinates can be directly provided as :class:`ndarray`.
+    order : array-like, length=3
+        The order of decreasing variance. Setting order to (2, 0, 1)
+        results in highest variance along the y-axis and lowest along
+        the x-axis. Default = (0, 1, 2)
+
+    Returns
+    -------
+    AtomArray or ndarray, shape=(n,3)
+        The atoms with coordinates centered at the orgin and aligned with
+        xyz axes.
+
+    See Also
+    --------
+    rotate_centered
+    rotate_about_axis
+
+    Examples
+    --------
+    Align principal components to xyz axes (default), or specify the order
+    of variance.
+
+    >>> print("original variance =", atom_array.coord.var(axis=0))
+    original variance = [26.517471 20.008837  9.324988]
+    >>> moved = orient_principal_components(atom_array)
+    >>> print("moved variance =", moved.coord.var(axis=0))
+    moved variance = [28.906227 18.495403  8.449674]
+    >>> # Note the increase in variance along the x-axis
+    >>> # Specifying the order keyword changes the orientation
+    >>> moved_z = orient_principal_components(atom_array, order=(2, 1, 0))
+    >>> print("moved (zyx) variance =", moved_z.coord.var(axis=0))
+    moved (zyx) variance = [ 8.449674 18.495403 28.906227]
```

Comment:
For the doctests, the values are rounded to 3 decimal places. With the rounded values your output passes the test.

```suggestion
    >>> print("original variance =", atom_array.coord.var(axis=0))
    original variance = [26.517 20.009  9.325]
    >>> moved = orient_principal_components(atom_array)
    >>> print("moved variance =", moved.coord.var(axis=0))
    moved variance = [28.906 18.495  8.450]
    >>> # Note the increase in variance along the x-axis
    >>> # Specifying the order keyword changes the orientation
    >>> moved_z = orient_principal_components(atom_array, order=(2, 1, 0))
    >>> print("moved (zyx) variance =", moved_z.coord.var(axis=0))
    moved (zyx) variance = [ 8.450 18.495 28.906]
```